Tyespec migration for  maps

welovej opened the pull request:
Parameter `privateEndpointConnectionName`  for  `PrivateEndpointConnections_Get`  `PrivateEndpointConnections_Create`  `PrivateEndpointConnections_Delete` has diffs with PrivateEndpointConnectionName   in common types. So we decide to add these below to mitigate:

```
@@maxLength(CommonTypes.PrivateEndpointConnection.name, 98);
@@minLength(CommonTypes.PrivateEndpointConnection.name, 3);
@@pattern(CommonTypes.PrivateEndpointConnection.name, "^[^%&:\\\\/#?]+$");
```

------------


This PR migrates your latest version (identified by the tag in your readme.md) of swagger to TypeSpec. We already tried our best to make sure the TypeSpec represents same as previous swagger. Since we lack the business knowledge, please validate this PR again to make sure it's functional equivalent as before. The local validation step is at [Getting started | TypeSpec Azure](https://azure.github.io/typespec-azure/docs/migrate-swagger/01-get-started/#review-and-adjust-the-typespec)

Besides, TypeSpec encourages to follow [ARM guidelines](https://aka.ms/rpguidelines). Therefore, some representations in your previous swagger will be fixed to follow these guidelines. When you see differences in your local validation, please keep this note in mind.

Please reach out to [TypeSpec Discussions Channel](https://teams.microsoft.com/l/channel/19%3a906c1efbbec54dc8949ac736633e6bdf%40thread.skype/TypeSpec%2520Discussion%2520%25F0%259F%2590%25AE?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) if there is any help needed.

welovej committed:
converted

welovej committed:
could compile

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Update routes.tsp

welovej committed:
update

welovej committed:
prettier

qiaozha reviewed (dismissed):
Please follow the new folder structure guideline and work with v-xuto

v-jiaodi committed:
Merge branch 'main' into TspMig-Maps

v-jiaodi committed:
update client name

welovej committed:
delete before relocate

welovej committed:
update

weidongxu-microsoft committed:
rename for java

welovej committed:
Update suppressions.yaml

ChenxiJiang333 committed:
Add Python support to Azure Maps Management Client

Added Python support for Azure Maps Management Client and updated related encryption identity models.

v-jiaodi committed:
Update tspconfig.yaml for compatibility and LRO

welovej committed:
update

v-jiaodi committed:
Remove compatibility-lro from tspconfig.yaml

Removed compatibility-lro setting from tspconfig.yaml

pshao25 committed:
Merge branch 'main' into TspMig-Maps

welovej committed:
update

welovej committed:
update
